src/utils.py

```diff
@@ -0,0 +1,14 @@
+def is_there_a_opposite_match_in_next_k_blocks(startblock, k, focus_pair,
+                                               swaps_by_block, sorted_blocks):
+    for j in range(startblock, startblock + k):
+        for o in swaps_by_block.get(sorted_blocks[j], []):
+            if focus_pair is not None:
+                if focus_pair[0] in o['buyToken'] and \
+                    (focus_pair[1] in o['sellToken'] or
+                     (len(focus_pair) >= 3 and focus_pair[2]
```

Comment:
How would a `focus_pair` be longer than 2?

Reply:
Here, I was checking whether WETH can be settled against either DAI, USDT or USDC. This assumes that we can exchange DAI, USDT or USDC basically without any slippage on curve.

But I deleted that part as it is only confusing and not important.